org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/LinkedResourcesUpdater.java

```diff
@@ -39,30 +47,30 @@
  * Note that currently, we only include linked resources that are folders.
  */
 final class LinkedResourcesUpdater {
+    private static final QualifiedName LINKED_RESOURCES_KEY = new QualifiedName(CorePlugin.PLUGIN_ID, "linkedResources");
```

Comment:
Persistent properties have a size limitation which means that if a user defines lots of linked resources, Buildship will fail with `org.eclipse.core.internal.resources.ResourceException: Could not set property: org.eclipse.buildship.core linkedResources. Value is too long` exception.

I think we should save the existing state into a properties file at the same place where we store the classpath container cache.